tests/test_validate_parameters.py

```diff
@@ -10,14 +10,47 @@
 
 import pytest
 import yaml
+from click.testing import CliRunner
 
+from reana_client.utils import cwl_load
 from reana_client.validation import (
     _validate_dangerous_operations,
     _validate_serial_parameters,
     _validate_yadage_parameters,
+    _validate_cwl_parameters,
 )
 
 
+def test_validate_parameters_cwl(
+    create_cwl_yaml_workflow_schema,
+    cwl_workflow_spec_one,
+    cwl_workflow_spec_two,
+    capsys,
+):
+    """Validate parameters for CWL workflows."""
+    runner = CliRunner()
+    with runner.isolated_filesystem():
+        with open("test.cwl", "w") as f:
+            f.write(cwl_workflow_spec_one)
+        reana_yaml = yaml.load(create_cwl_yaml_workflow_schema, Loader=yaml.FullLoader)
+        reana_yaml["workflow"]["specification"] = cwl_load(
+            reana_yaml["workflow"].get("file")
+        )
+        _validate_cwl_parameters(reana_yaml)
+        captured = capsys.readouterr()
+        assert not captured.out
+
+    # Wrong Input parameter used
+    with runner.isolated_filesystem():
+        with open("test.cwl", "w") as f:
+            f.write(cwl_workflow_spec_two)
+        reana_yaml = yaml.load(create_cwl_yaml_workflow_schema, Loader=yaml.FullLoader)
+        reana_yaml["workflow"]["specification"] = cwl_load(
+            reana_yaml["workflow"].get("file")
+        )
+        pytest.raises(SystemExit, _validate_cwl_parameters, reana_yaml)
```

Comment:
Could you add a check of the error output and verify which parameters are wrong?

E.g. with this output

```console
$ cwltool --validate --strict workflow/cwl/helloworld.cwl 
INFO /Users/marco/.virtualenvs/r/bin/cwltool 1.0.20191022103248
INFO Resolved 'workflow/cwl/helloworld.cwl' to 'file:///Users/marco/code/reanahub/reana-demo-helloworld/workflow/cwl/helloworld.cwl'
ERROR Tool definition failed validation:
workflow/cwl/helloworld.cwl:34:5: Step is missing required parameter 'sleeptim'
```

verify  `sleeptim`  is in the error output.

Reply:
This is not addressed.